Reduce Thomas and Fotis code ownership

thomasspriggs opened the pull request:
This PR removes my code ownership except for that of the incremental SMT2 decision procedure. I am expecting the time I can make available for code reviews to be rather limited in the immediate future. As I do not wish to hold up anyone else's work waiting for my review, I am reducing the number of areas for which I am codeowner. I intend to retain code ownership of the incremental SMT2 decision procedure as I have particular expertise in this area.

Included in this PR is a similar change for Fotis.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

thomasspriggs committed:
Remove Thomas Spriggs codeownership except for incremental SMT2

I am expecting the time I can make available for code reviews to be
rather limited in the immediate future. As I do not wish to hold up
anyone elses work waiting for my review, I am reducing the number of
areas for which I am codeowner. I intend to retain code ownership of the
incremental SMT decision procedure as I have particular expertise in
this area.

martin-cs reviewed (approved):
Thanks for all of your hard work.

NlightNFotis committed:
Remove @NlightNFotis from `CODEOWNERS`.

With my tenure at Diffblue coming to an end, and a vision of myself
working in different domains going forward, I don't think there's
going to be much time to spare working on CBMC on the side. To that
end, I believe it's fair to the project to remove myself from the
`CODEOWNERS` file so that nothing is pending my approval in the future.

It has been an honour and a pleasure working on the project, and
I'm happy with what we've achieved these years.
